namer/namer.cc

```diff
@@ -694,6 +697,13 @@ class SymbolFinder {
             return tree;
         }
 
+        if (inBlock) {
+            if (auto e = ctx.state.beginError(core::Loc(ctx.file, lhs->loc),
+                                              core::errors::Namer::DynamicConstantAssignment)) {
+                e.setHeader("Dynamic constant assignment");
```

Comment:
Let's make this error say something like

```
Assigning constant `{}` inside a block
```

(This is more clear what the cause of the problem is and includes the name of the offending constant, which helps people locate it more quickly without having to read line numbers.)

and potentially use `addErrorNote` to suggest people move the constant assignment outside of the block.